python/cli/prelude_cli/views/build.py

```diff
@@ -188,3 +190,59 @@ def delete_threat(controller, threat, purge):
     with Spinner(description='Removing threat'):
         data = controller.delete_threat(threat_id=threat, purge=purge)
     print_json(data=data)
+
+
+@build.command('create-detection')
+@click.argument('partner', type=click.Choice([Control.CROWDSTRIKE.name], case_sensitive=False))
+@click.option('-r', '--rules', help='CRWD detection rules, as string-formatted json', default=None, type=str)
+@click.option('--rules_file', help='CRWD detection rules, from a json file', default=None, type=click.File())
+@click.option('-t', '--test', help='ID of the test this detection is for', required=True, type=str)
+@click.option('--detection_id', help='detection ID', default=None, type=str)
+@click.option('--rule_id', help='rule ID', default=None, type=str)
+@click.pass_obj
+@handle_api_error
+def create_detection(controller, rules, rules_file, test, partner, detection_id, rule_id):
+    """ Create an EDR detection rule """
+    if (rules and rules_file) or (not rules and not rules_file):
+        raise UsageError('Exactly one of --rules and --rules_file must be set')
+
+    with Spinner(description='Creating new detection'):
+        t = controller.create_detection(
+            rules=json.loads(rules) if rules else json.load(rules_file),
+            test_id=test,
+            partner=Control[partner],
+            detection_id=detection_id,
+            rule_id=rule_id
+        )
+    print_json(data=t)
+
+
+@build.command('update-detection')
+@click.argument('detection')
+@click.option('-r', '--rules', help='CRWD detection rules, as string-formatted json', default=None, type=str)
```

Comment:
should update also accept a rules_file to be consistent with create?

Reply:
yes. done!